### PR TITLE
[MSFT] [Design Partitioning] Bubble up wire manipulation ops also

### PR DESCRIPTION
**Pull request by teqdruid**
Towards #2365: Automatically tag "wire manipulation" ops. This is easier than doing this in the wire cleanup pass it turns out.

### CODE REVIEW

File: test/Dialect/MSFT/partition.mlir [hunk 1]
```diff
@@ -1,4 +1,3 @@
-// RUN: circt-opt %s --msft-partition -verify-diagnostics -split-input-file | FileCheck %s
```
Comment (teqdruid) [inline test/Dialect/MSFT/partition.mlir:1]: The output from "--msft-partition" without wire cleanup is sooo complex so as to make visual verification impossible, so testing it with filecheck is worthless given that I would just be copy and pasting it.

Comment (mikeurbach) [reply test/Dialect/MSFT/partition.mlir:1]: I think its possible to write a meaningful FileCheck test for just a small part of the design... but testing both of these passes in conjunction sounds fine to me.

**Review by mikeurbach — APPROVED** [review]
Makes sense to me.

File: test/Dialect/MSFT/partition.mlir (hunk 1 above)
Comment (teqdruid) [reply test/Dialect/MSFT/partition.mlir:1]: Yeah, perhaps just one op tagged.

**Comment by teqdruid** [comment]
Thanks for the review @mikeurbach !

**Pull request merged**